control-plane/pkg/reconciler/sink/auth_provider.go

```diff
@@ -24,13 +24,16 @@ type SecretLocator struct {
 	*eventing.KafkaSink
 }
 
-func (ks *SecretLocator) SecretName() (string, bool, error) {
+func (ks *SecretLocator) SecretName() (string, bool) {
 	if !ks.Spec.HasAuthConfig() {
-		return "", false, nil
+		return "", false
 	}
-	return ks.Spec.Auth.Secret.Ref.Name, true, nil
+	return ks.Spec.Auth.Secret.Ref.Name, true
 }
 
-func (ks *SecretLocator) SecretNamespace() (string, bool, error) {
-	return ks.Namespace, true, nil
+func (ks *SecretLocator) SecretNamespace() (string, bool) {
+	if !ks.Spec.HasAuthConfig() {
+		return "", false
+	}
+	return ks.Namespace, true
```

Comment:
what about this?
```suggestion
	return ks.Namespace, ks.Spec.HasAuthConfig()
```

Reply:
I can revert that one, not a big issue.

I thought:
- If there's no secret specified in the spec, we should also make `SecretNamespace()` return nil, similar to `SecretName()`

In the end, the `SecretName()` is always called before `SecretNamespace()` and the execution stops before `SecretNamespace()` call, but I like consistency.

I don't have a strong opinion and can revert

Reply:
Oh, you have a suggestion. That's actually better. Committing that now